### PR TITLE
[Doc] Add sections about how to enable and configure chunking for java client producers/consumers

### PR DESCRIPTION
**Pull request by momo-jun**
### Motivation
To enhance the [Pulsar Feature Matrix](https://docs.google.com/spreadsheets/d/1YHYTkIXR8-Ql103u-IMI18TXLlGStK8uJjDsOOA0T20/edit#gid=1784579914), add the required document for chunking in the java client category.

### Modifications
1. Add a section about how to enable message chunking for java client producers, and add one parameter.
image
image

2. Add a section about how to configure message chunking for java client consumers, and add three parameters.
image
image

### Documentation

- [x] `doc`

### CODE REVIEW

**Comment by momo-jun** [comment]
Hi @RobertIndie @Anonymitaet , can you please review this doc PR? Thank you.

**Review by codelipenghui — APPROVED** [review]

File: site2/docs/client-libraries-java.md [hunk 1]
```diff
@@ -338,6 +354,9 @@ When you create a consumer, you can use the `loadConf` configuration. The follow
 `replicateSubscriptionState`|boolean|If `replicateSubscriptionState` is enabled, a subscription state is replicated to geo-replicated clusters.|false
 `negativeAckRedeliveryBackoff`|RedeliveryBackoff|Interface for custom message is negativeAcked policy. You can specify `RedeliveryBackoff` for a consumer.| `MultiplierRedeliveryBackoff`
 `ackTimeoutRedeliveryBackoff`|RedeliveryBackoff|Interface for custom message is ackTimeout policy. You can specify `RedeliveryBackoff` for a consumer.| `MultiplierRedeliveryBackoff`
+`autoAckOldestChunkedMessageOnQueueFull`|boolean|Whether to automatically acknowledge pending chunked messages when the threashold of `maxPendingChunkedMessage` is reached. If set to `false`, these messages will be redelivered by their broker. |true
+`maxPendingChunkedMessage`|int| The maximum size of a queue holding pending chunked messages. When the threshold is reached, the consumer drops pending messages to optimize memory utilization.|10
+`expireTimeOfIncompleteChunkedMessageMillis`|long|The time interval to expire incomplete chunks if a consumer fails to receive all the chunks in the specified time period. The default value is 1 minute. | 1
```
Comment (RobertIndie) [inline site2/docs/client-libraries-java.md:359]:
```suggestion
`expireTimeOfIncompleteChunkedMessageMillis`|long|The time interval to expire incomplete chunks if a consumer fails to receive all the chunks in the specified time period. The default value is 1 minute. | 600000
```

Comment (RobertIndie) [reply site2/docs/client-libraries-java.md:359]: `expireTimeOfIncompleteChunkedMessageMillis ` here is in milliseconds unit. The default value is 60000(which is 1 min).

**Review by RobertIndie — CHANGES_REQUESTED** [review]

File: site2/docs/client-libraries-java.md (hunk 1 above)
Comment (momo-jun) [reply site2/docs/client-libraries-java.md]: Thanks for pointing this out. I've corrected it. PTAL.

**Review by RobertIndie — APPROVED** [review]

File: site2/docs/client-libraries-java.md [hunk 2]
```diff
@@ -404,6 +423,22 @@ consumer.acknowledge(messages)
 >     .build();
 > ```
 
+### Configure chunking
+
+You can limit the maximum number of chunked messages a consumer maintains concurrently by configuring the `maxPendingChunkedMessage` and `autoAckOldestChunkedMessageOnQueueFull` parameter. When the threshold is reached, the consumer drops pending messages by silently acknowledging them or asking the broker to redeliver them later. The `expireTimeOfIncompleteChunkedMessage` parameter decides the time interval to expire incomplete chunks if the consumer fails to receive all chunks of a message within the specified time period.
```
Comment (BewareMyPower) [inline site2/docs/client-libraries-java.md:428]:
```suggestion
You can limit the maximum number of chunked messages a consumer maintains concurrently by configuring the `maxPendingChunkedMessage` and `autoAckOldestChunkedMessageOnQueueFull` parameters. When the threshold is reached, the consumer drops pending messages by silently acknowledging them or asking the broker to redeliver them later. The `expireTimeOfIncompleteChunkedMessage` parameter decides the time interval to expire incomplete chunks if the consumer fails to receive all chunks of a message within the specified time period.
```

Comment (momo-jun) [reply site2/docs/client-libraries-java.md]: Sharp eyes! Thanks for pointing this out. I've corrected it. PTAL.

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**